### ISSUE TITLE
Webadmin 404 error in 19.09

### ISSUE THREAD

**Issue by vampik** [processone/ejabberd#3050]
## Environment

- ejabberd version: 19.09
- Erlang version: `Erlang (SMP,ASYNC_THREADS,HIPE) (BEAM) emulator version 10.4.4`
- OS: Linux (Gentoo)
- Installed from: other

## Configuration

```yaml
hosts:
  - "example.net"
listen:
...
  -
    request_handlers:
      "/admin": ejabberd_web_admin
    port: 5281
    ip: "::"
    module: ejabberd_http
    tls: true
    protocol_options:
      - "no_sslv3"
      - "no_tlsv1"
      - "cipher_server_preference"
      - "no_compression"
    dhfile: "/etc/ssl/ejabberd/dhparams.pem"
    ciphers: "ECDH:DH:!3DES:!aNULL:!eNULL:!MEDIUM@STRENGTH"
...
```
 
## Errors from error.log/crash.log

No errors

## Bug description

This seems to be related to #2989, but the conversation is closed there, so I open a new issue.

Since 19.08 accessing https://jabber.example.net:5281/admin gave me "Host not served" error. The host I serve is example.net, but the fqdn of the ejabberd server itself is jabber.example.net.

I've read that it should be fixed in 19.09, but it is not.

Now accessing https://jabber.example.net:5281/admin gives me "404 Not Found" error instead.

**Comment by giostark** [comment]
Hi vampik,
(I'm a noob , so please be patient)
Im facing this issue too. Cant login in the webadmin anymore. 
My web interface before the issue come at http://opendimensions.ddns.net:5280/admin
I use a freeDNS service for route the request in my NAS. 
What they are suggesting as solution is change the domain name in such http://**xmpp**.opendimensions.ddns.net:5280/admin ? (modifying also the host in the config file as **xmpp**.opendimensions.ddns.net)
I use Arch Linux , and I cant understand why it stop to work. With the 19.05 worked till 2 day ago. 
After update Arch it stop to work (on PC guest so not the XMPP server that is in a NAS) . So I supposed was a PC side issue. Then I updated the XMPP server at 19.09 but still no luck. I tried to login from windows but nothing. 
The netstat -lntu report the port 5280 in linsten mode. (I didn touch any config files)

**Comment by zinid** [comment]
Yeah, I confirm the bug remains. Seems like only webadmin is affected.

**Comment by vladviolenty** [comment]
Confirm this bug in example config file
http://IP:5280/admin - 404 not found
http://IP:5280/admin/vhosts - 404 not found
http://IP:5280/admin/server/HOST - work fine

**Comment by giostark** [comment]
> http://IP:5280/admin/server/HOST - work fine
Yep , work just with the ip. 
With the hostname (DNS route) not work.
Also I had to use the full admin nick (admin@opendimensions.ddns.net). Before was enough "admin".

**Comment by NohaHammaK** [comment]
I am on windows 10 and I have the same bug, 19.09 gives 404 not found, 19.08 gives Host not served, previous versions fail during installation for me.

**Comment by prefiks** [comment]
This should be fixed by commit 8d571adca845285aa4928ada25d31cdbc32ef0a0, in this version check for valid host will be only performed if host was not passed in auth data.
